src/components/menu/__internal__/keyboard-navigation/index.js

```diff
@@ -1,11 +1,8 @@
 import Events from "../../../../utils/helpers/events";
 import MenuItem from "../../menu-item";
 
-function characterNavigation(event, focusableItems, currentFocusedIndex) {
-  event.stopPropagation();
-  let firstMatch;
-  let nextMatch;
-  const selectedKey = event.key.toLowerCase();
+function characterNavigation(inputString, focusableItems, currentFocusedIndex) {
+  if (!inputString || inputString === "") return currentFocusedIndex;
```

Comment:
I think this could just be as `""` is `falsy`
```suggestion
  if (!inputString) return currentFocusedIndex;
```